src/components/views/rooms/MessageComposer.tsx

```diff
@@ -308,7 +308,12 @@ export default class MessageComposer extends React.Component<IProps, IState> {
     };
 
     private updateRecordingState() {
-        this.voiceRecording = VoiceRecordingStore.instance.getActiveRecording(this.props.room.roomId);
+        this.voiceRecording =
+            this.props.relation?.rel_type === ("io.element.thread")
+            || this.props.relation?.rel_type === RelationType.Thread ?
+                VoiceRecordingStore.instance.getActiveRecording(this.props.room.roomId+this.props.relation.event_id)
+                :
+                VoiceRecordingStore.instance.getActiveRecording(this.props.room.roomId);
```

Comment:
It would be great if you could clean up this, as you use it in many places, and it lends itself well to being turned into a function.

Something like

```typescript
function getVoiceRecordingId(room: Room, relation?: IEventRelation): string {
    if (relation?.rel_type === "io.element.thread" || relation?.rel_type === RelationType.Thread) {
      return room.roomId+relation.event_id;
    } else {
      return room.roomId;
    }
}
```

And then at the usage places
```typescript
this.voiceRecording = VoiceRecordingStore.instance.getActiveRecording(getVoiceRecordingId(this.props.room, this.props.relation));
```

Reply:
Additionally, it’d be IMO even better if you’d introduce some kind of separator between the roomId and eventId to prevent potential overlaps should the roomIds ever get longer in the future.